CMakeCommon.cmake

```diff
@@ -246,3 +251,42 @@ function (GenerateAddOnProject target acVersion devKitDir addOnName addOnSources
     SetCompilerOptions (${target} ${acVersion})
 
 endfunction ()
+
+function (check_valid_language_code configFile languageCode)
+    file (READ ${configFile} configsContent)
+    string (JSON configuredLanguagesList GET "${configsContent}" "languages")
+    string (JSON configuredLanguagesListLen LENGTH "${configsContent}" "languages")
+    set (i 0)
+    while (${i} LESS ${configuredLanguagesListLen})
+        string (JSON language GET "${configuredLanguagesList}" ${i})
+        if (${language} STREQUAL ${languageCode})
+            return ()
+        endif ()
+        math (EXPR i "${i} + 1")
+    endwhile()
+
+    message (FATAL_ERROR "Language code ${languageCode} is not part of the configured languages in ${configFile}.")
+endfunction ()
+
+function (verify_api_devkit_folder devKitPath)
+    if (NOT EXISTS ${devKitPath})
+        message (FATAL_ERROR "The supplied API DevKit path ${devKitPath} does not exist")
+    endif ()
+
+    cmake_path (GET devKitPath FILENAME currentFolderName)
+    if (NOT "${currentFolderName}" STREQUAL "Support")
```

Comment:
This is an example where quoting is significant in the conditional syntax. `if()` and `while()` are special in that they see quotes around their arguments. In this case, quoting causes the RHS to be observed as a string, despite being `<variable|string>`. This is the desirable behavior here, just wanted to bring attention to it. For more, see [the `if()` docs](https://cmake.org/cmake/help/latest/command/if.html#variable-expansion) and the policy mentioned in it.

Reply:
Thanks for the explanation! I did not know about this and so the clarifications are very much appreciated.